lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -429,11 +433,55 @@ function dosomething_user_validate_address_field($form, &$form_state) {
     }
     // We have a full address, save it!
     else {
-      dosomething_user_set_address_values($form, $form_state, $formatted_address);
+      if ($form['#form_id'] === 'dosomething_user_info_form') {
+        dosomething_user_set_address_values_from_info_form($form, $form_state, $formatted_address);
+      }
+      else {
+        dosomething_user_set_address_values_from_edit_form($form, $form_state, $formatted_address);
+      }
     }
```

Comment:
@chloealee yeah, but `dosomething_user_validate_address_field` is meant to deal with the actual `address_field` type and you have to do a lot of work to turn the manually created form items into the type to match the address field format, which is probably why it isn't saving/updating